backend/pipeline/piepline_api_execution_views.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from typing import Any
+
+from pipeline.deployment_helper import DeploymentHelper
+from pipeline.models import Pipeline
+from rest_framework import status, views
+from rest_framework.request import Request
+from rest_framework.response import Response
+from scheduler.tasks import execute_pipeline_task
+
+logger = logging.getLogger(__name__)
+
+
+class PipelineApiExecution(views.APIView):
+    def initialize_request(
+        self, request: Request, *args: Any, **kwargs: Any
+    ) -> Request:
+        """To remove csrf request for public API.
+
+        Args:
+            request (Request): _description_
+
+        Returns:
+            Request: _description_
+        """
+        setattr(request, "csrf_processing_done", True)
+        return super().initialize_request(request, *args, **kwargs)
+
+    @DeploymentHelper.validate_api_key
+    def post(
+        self, request: Request, org_name: str, pipeline_id: str, pipeline: Pipeline
+    ) -> Response:
+        execute_pipeline_task.delay(
+            workflow_id="",
+            org_schema=org_name,
+            execution_action="",
+            execution_id="",
```

Comment:
@muhammad-ali-e why is workflow and execution ID empty?

Reply:
I don't know the requirement of those feilds ? Because executionId will create on the fly, And workflow_id will be get from Pipeline entity

Reply:
Okay, if that's the case then we can remove these params from `execute_pipeline_task` right? Was it from the original implementation of call the WF execution twice to send FE logs?
Anyway, please mark it with a TODO to review this portion

Reply:
Already have a TODO there, It required Data migrations